frontend/src/domain/entities/vessel/track/index.test.ts

```diff
@@ -0,0 +1,276 @@
+    expect(secondLineFeature!.speed).toEqual(2.9)
+    expect(secondLineFeature!.isTimeEllipsis).toEqual(false)
+    expect(secondLineFeature!.trackType!.code).toEqual('TRANSIT')
+    expect(secondLineFeature!.getId()).toEqual('vessel_track:VESSEL_ID:line:1')
+    expect(secondLineFeature!.getGeometry()?.getLength().toString()).toContain('8439')
+    // A LineString geometry is of type Coordinate[][]
+    expect(secondLineFeature!.getGeometry()?.getCoordinates()).toEqual([
+      [-372586.33568508667, 5849526.831117608],
+      [-379710.7830958562, 5854050.285232945]
+    ])
+
+    // Last position
+    const positionFeature = positionFeatures[5] as VesselPointFeature
+    expect(positionFeature!.course).toEqual(263)
+    expect(positionFeature!.getId()).toEqual('vessel_track:VESSEL_ID:position:5')
+    // A Point geometry is of type Coordinate[]
+    expect(positionFeature!.getGeometry()?.getCoordinates()).toEqual([-402308.6397268907, 5852757.632510743])
+  })
+
+  it('getTrackType Should return FISHING When two positions have the isFishing property set as true', async () => {
+    // Given
+    const firstPosition = DUMMY_VESSEL_TRACK[0] as VesselPosition
+    firstPosition.isFishing = true
+    const secondPosition = DUMMY_VESSEL_TRACK[1] as VesselPosition
+    secondPosition.isFishing = true
+
+    // When
+    const trackType = getTrackType([firstPosition, secondPosition], false)
+
+    // Then
+    expect(trackType.code).toEqual('FISHING')
+  })
+
+  it('getTrackType Should not return FISHING When first position does have the isFishing property set as true', async () => {
+    // Given
+    const firstPosition = DUMMY_VESSEL_TRACK[0] as VesselPosition
+    firstPosition.isFishing = false
+    const secondPosition = DUMMY_VESSEL_TRACK[1] as VesselPosition
+    secondPosition.isFishing = true
+
+    // When
+    const trackType = getTrackType([firstPosition, secondPosition], false)
+
+    // Then
+    expect(trackType.code).toEqual('TRANSIT')
+  })
+
+  it('getTrackType Should not return FISHING When second position does have the isFishing property set as true', async () => {
+    // Given
+    const firstPosition = DUMMY_VESSEL_TRACK[0] as VesselPosition
+    firstPosition.isFishing = true
+    const secondPosition = DUMMY_VESSEL_TRACK[1] as VesselPosition
+    secondPosition.isFishing = false
+
+    // When
+    const trackType = getTrackType([firstPosition, secondPosition], false)
+
+    // Then
+    expect(trackType.code).toEqual('TRANSIT')
+  })
+
+  it('getTrackType Should return FISHING When only one position is given and does have the isFishing property set as true', async () => {
+    // Given
+    const firstPosition = DUMMY_VESSEL_TRACK[0] as VesselPosition
+    firstPosition.isFishing = true
+
+    // When
+    const trackType = getTrackType([firstPosition], false)
+
+    // Then
+    expect(trackType.code).toEqual('FISHING')
+  })
+})
+
+const DUMMY_VESSEL_TRACK = [
```

Comment:
In Jest, the convention is to move both stubs and mocks inside a `__tests__/__mocks__` folder.